common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockHandler.java

```diff
@@ -218,28 +218,33 @@ public ShuffleMetrics() {
     private final int shuffleId;
     // An array containing mapId and reduceId pairs.
     private final int[] mapIdAndReduceIds;
+    private final int stageAttemptId;
 
     ManagedBufferIterator(String appId, String execId, String[] blockIds) {
       this.appId = appId;
       this.execId = execId;
       String[] blockId0Parts = blockIds[0].split("_");
-      if (blockId0Parts.length != 4 || !blockId0Parts[0].equals("shuffle")) {
-        throw new IllegalArgumentException("Unexpected shuffle block id format: " + blockIds[0]);
-      }
+      checkBlockId(blockId0Parts, blockIds, 0);
```

Comment:
What about merging the two lines above into one function? It can be named like `splitBlockId()`and its header would be a bit a bit simpler:

```java
String[] splitBlockId(String blockId)
```

This way the `index` and `blockIdParts` should not be coordinated.

Reply:
Thanks, that's more coding clean, done in 012ae1d.